hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyDeletingService.java

```diff
@@ -55,17 +58,21 @@
   // The thread pool size for key deleting service.
   private final static int KEY_DELETING_CORE_POOL_SIZE = 2;
 
+  private final OzoneManager ozoneManager;
   private final ScmBlockLocationProtocol scmClient;
   private final KeyManager manager;
+  private ClientId clientId = ClientId.randomId();
   private final int keyLimitPerTask;
   private final AtomicLong deletedKeyCount;
   private final AtomicLong runCount;
 
-  public KeyDeletingService(ScmBlockLocationProtocol scmClient,
+  public KeyDeletingService(OzoneManager ozoneManager,
+      ScmBlockLocationProtocol scmClient,
       KeyManager manager, long serviceInterval,
       long serviceTimeout, Configuration conf) {
     super("KeyDeletingService", serviceInterval, TimeUnit.MILLISECONDS,
         KEY_DELETING_CORE_POOL_SIZE, serviceTimeout);
+    this.ozoneManager = ozoneManager;
```

Comment:
Can you think of a way to avoid passing `OzoneManager` to this class? We now have a low level class depending on a higher level class and potentially a circular dependency.

Reply:
Not a blocker, can be handled separately.

Reply:
For this, we can use Java 8 method reference. 
And also it will be easy for UT's also for testing.